config/config.go

```diff
@@ -344,6 +344,24 @@ func (c *Configuration) EnsureFilePermissions() error {
 
 	r := regexp.MustCompile("^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}$")
 
+	// add trailing slash on data directory is no trailing slash exists
+	if ! strings.HasSuffix(c.System.Data, "/") {
+		c.System.Data = c.System.Data + "/"
+	}
+
+	// create the daemon-data dir if it doesn't exist
+	p, _ := path.Split(c.System.Data)
+
+	if _, err := os.Stat(c.System.Data); err != nil {
```

Comment:
This needs to report an error not caused by `IsNotExist`

Reply:
The Split does not return an error but a path and filename. As we are expecting a path I just throw out the blank file name.

 As for the Stat I am adding a return for the error.